datafusion/common/src/scalar/mod.rs

```diff
@@ -3648,6 +3648,12 @@ fn fmt_list(arr: ArrayRef, f: &mut fmt::Formatter) -> fmt::Result {
 
 impl fmt::Debug for ScalarValue {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        fn fmt_binary(data: &[u8]) -> String {
```

Comment:
I think this allocates a new string for each byte... Perhaps it would be better if this called `write!` directly on the formatter. Something like



```suggestion
        fn fmt_binary(data: &[u8], f: &mut fmt::Formatter) -> fmt::Result {
```